kernel/memory_structs/src/lib.rs

```diff
@@ -392,17 +533,12 @@ macro_rules! implement_page_frame_range {
                         
             #[doc = "A range of [`" $chunk "`]s that are contiguous in " $desc " memory."]
             #[derive(Clone, PartialEq, Eq)]
-            pub struct $TypeName(RangeInclusive<$chunk>);
+            pub struct $TypeName<P: PageSize = Page4KiB>(RangeInclusive<$chunk::<P>>);
 
             impl $TypeName {
-                #[doc = "Creates a new range of [`" $chunk "`]s that spans from `start` to `end`, both inclusive bounds."]
-                pub const fn new(start: $chunk, end: $chunk) -> $TypeName {
-                    $TypeName(RangeInclusive::new(start, end))
-                }
-
                 #[doc = "Creates a `" $TypeName "` that will always yield `None` when iterated."]
                 pub const fn empty() -> $TypeName {
-                    $TypeName::new($chunk { number: 1 }, $chunk { number: 0 })
+                    $TypeName::new($chunk { number: 11, size: PhantomData }, $chunk { number: 0, size: PhantomData })
```

Comment:
why `11` instead of `1`? what's the difference?

Reply:
That wasn't intentional; I'll remove it. Must have been mistakenly added when adding `size: PhantomData` to everything.